cl/search/api_serializers.py

```diff
@@ -289,3 +289,52 @@ def get_fields(self):
                 fields.pop(field)
         fields = OrderedDict(sorted(fields.items()))  # Sort by key
         return fields
+
+
+class SearchESResultSerializer(serializers.Serializer):
+    """The serializer for Elasticsearch results.
+    Does not presently support the fields argument.
+    """
+
+    def update(self, instance, validated_data):
+        raise NotImplementedError
+
+    def create(self, validated_data):
+        raise NotImplementedError
+
+    es_field_mappings = {
+        "boolean": serializers.BooleanField,
+        "text": serializers.CharField,
+        "keyword": serializers.CharField,
+        "date": serializers.DateTimeField,
+        # Numbers
+        "integer": serializers.IntegerField,
+    }
+    skipped_fields = ["text", "sha1", "docket_slug"]
+
+    def get_fields(self):
+        """Return a list of fields so that they don't have to be declared one
+        by one and updated whenever there's a new field.
+        """
+        fields = {
+            "snippet": serializers.CharField(read_only=True),
+            "panel_ids": serializers.ListField(read_only=True),
+        }
+
+        properties = self._context["schema"]["oral_arguments"]["mappings"][
+            "properties"
+        ]
```

Comment:
Yeah, this `_context` is defined in `SearchViewSet`:

```
serializer = SearchESResultSerializer(
                    result_page,
                    many=True,
                    context={
                        "schema": AudioDocument._index.get_mapping()[
                            "oral_arguments"
                        ]["mappings"]
                    },
                )
```

This is the index mapping so it returns a list of fields available in the index.
I changed it a bit in order to check the fields as:
`properties = self._context["schema"]["properties"]`
 
So it doesn't use a specific model key like it was `oral_arguments`, so this can be easily reused when moving other models to ES.